cmd/transaction.go

```diff
@@ -39,12 +42,65 @@ func transactionCmdChain() *cobra.Command {
 			return nil
 		},
 	}
-	transactionLookupCmd.Flags().String(transactionid, "", "id of the transaction being looked up")
-	transactionLookupCmd.MarkFlagRequired(transactionid)
+	transactionLookupCmd.Flags().String(transactionId, "", "id of the transaction being looked up")
+	transactionLookupCmd.MarkFlagRequired(transactionId)
 
-	transactionCallCmd := &cobra.Command{}
-	transactionContractCmd := &cobra.Command{}
+	transactionCallCmd := &cobra.Command{
+		Use:   "call",
+		Short: "call functions on a mazzaroth channel",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := mazzaroth.NewMazzarothClient(mazzaroth.WithAddress(viper.GetString(channelAddress)))
+			if err != nil {
+				return err
+			}
+			pk, err := crypto.FromHex(viper.GetString(privateKey))
+			if err != nil {
+				return err
+			}
+			sender, err := xdr.IDFromHexString(viper.GetString(publicKey))
+			if err != nil {
+				return err
+			}
+
+			channelId, err := xdr.IDFromHexString(viper.GetString(channelId))
+			if err != nil {
+				return err
+			}
+			xdrArgs := make([]xdr.Argument, 0, 0)
+			values := viper.GetStringSlice(arguments)
+			for _, a := range values {
+				xdrArgs = append(xdrArgs, xdr.Argument(a))
+			}
+
+			tx, err := mazzaroth.Transaction(sender, channelId).Call(mazzaroth.GenerateNonce(), defaultBlockExpirationNumber).Function(viper.GetString(function)).Arguments(xdrArgs...).Sign(pk)
+			if err != nil {
+				return err
+			}
+
+			id, rcpt, err := client.TransactionSubmit(cmd.Context(), tx)
+			if err != nil {
+				return err
+			}
+
+			if rcpt != nil {
+				jsonRcpt, err := json.MarshalIndent(rcpt, "", "\t")
+				if err != nil {
+					return err
+				}
+				fmt.Println("receipt object:")
+				fmt.Println(jsonRcpt)
```

Comment:
This needs to be converted to a string for the print: `fmt.Println(string(jsonRcpt))`

Reply:
pushed